lib/TikZImage.pm

```diff
@@ -156,60 +157,87 @@ sub draw {
-					" $working_dir/image.pdf $working_dir/image.svg > /dev/null 2>&1";
-			}
-		} elsif ($ext ne 'pdf') {
-			system WeBWorK::PG::IO::externalCommand('convert') .
-				join('',map {" -$_ " . $self->convertOptions->{input}->{$_}} (keys %{$self->convertOptions->{input}})) .
-				" $working_dir/image.pdf" .
-				join('',map {" -$_ " . $self->convertOptions->{output}->{$_}} (keys %{$self->convertOptions->{output}})) .
-				" $working_dir/image.$ext > /dev/null 2>&1";
+	# Create either one or two tex files with one small difference:
+	# set pgfsysdriver to pgfsys-dvisvgm.def for a tex file that dvisvgm will use
+	# Then make only the dvi, only the pdf, or both in case we are making tgz with svg via dvisvgm
+	if (($ext eq 'svg' || $ext eq 'tgz') && $svgMethod eq 'dvisvgm') {
+		open($fh, ">", "$working_dir/image-dvisvgm.tex")
+			or warn "Can't open $working_dir/image-dvisvgm.tex for writing.";
+		my @header = $self->header;
+		splice @header, 1, 0, "\\def\\pgfsysdriver{pgfsys-dvisvgm.def}\n";
+		print $fh @header;
+		print $fh $self->tex =~ s/\\\\/\\/gr . "\n";
+		print $fh $self->footer;
+		close $fh;
+		system "cd $working_dir && $latex image-dvisvgm.tex > latex.stdout 2> /dev/null && mv image-dvisvgm.dvi image.dvi";
+		chmod(0777, "$working_dir/image.dvi");
+	}
+	if ($ext ne 'svg' || ($ext eq 'svg' && $svgMethod ne 'dvisvgm')) {
+		open($fh, ">", "$working_dir/image.tex")
+			or warn "Can't open $working_dir/image.tex for writing.";
+		print $fh $self->header;
+		print $fh $self->tex =~ s/\\\\/\\/gr . "\n";
+		print $fh $self->footer;
+		close $fh;
+		system "cd $working_dir && $pdflatex image.tex > pdflatex.stdout 2> /dev/null";
+		chmod(0777, "$working_dir/image.pdf");
+	}
+
+	# Make derivatives of the dvi
+	if (($ext eq 'svg' || $ext eq 'tgz') && $svgMethod eq 'dvisvgm') {
+		if (-r "$working_dir/image.dvi") {
+			$self->use_svgMethod($working_dir, "image.dvi");
+		} else {
+			$self->warnLaTeXOutput($working_dir, "image.dvi");
 		}
+	}
 
-		if (-r "$working_dir/image.$ext") {
-			# Read the generated image file into memory
-			open(my $in_fh,  "<", "$working_dir/image.$ext")
-				or warn "Failed to open $working_dir/image.$ext for reading.", return;
-			local $/;
-			$data = <$in_fh>;
-            close($in_fh);
+	# Make derivatives of the pdf
+	if (($ext eq 'svg' || $ext eq 'tgz') && $svgMethod ne 'dvisvgm') {
+		if (-r "$working_dir/image.pdf") {
+			$self->use_svgMethod($working_dir, "image.pdf");
 		} else {
-			warn "Convert operation failed.";
+			$self->warnLaTeXOutput($working_dir, "image.pdf");
 		}
-	} else {
-		warn "File $working_dir/image.$tex_ext was not created.";
-		if (open(my $err_fh, "<", "$working_dir/pdflatex.stdout")) {
-			while (my $error = <$err_fh>) {
-				warn $error;
-			}
-			close($err_fh);
+	}
+	if ($ext eq 'tgz') {
+		if (-r "$working_dir/image.pdf") {
+			$self->use_convert($working_dir, "image.pdf", "png");
+		} else {
+			$self->warnLaTeXOutput($working_dir, "image.pdf");
 		}
 	}
+	elsif ($ext ne 'svg') {
```

Comment:
This should be
```suggestion
	elsif ($ext ne 'svg' && $ext ne 'pdf') {
```
No need to convert "image.pdf" to "image.pdf".   Furthermore this actually reduces the quality of the pdf image.

Reply:
Oops, I accidentally staged and committed this a few items back.